src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js

```diff
@@ -1,210 +1,92 @@
-import React, { useState, useEffect } from 'react'
-import { compose } from 'redux'
-import { graphql } from 'react-apollo'
+import React from 'react'
 import PropTypes from 'prop-types'
 import FormikSelect from '../../../../../components/formik-santiment-ui/FormikSelect'
-import FormikInput from '../../../../../components/formik-santiment-ui/FormikInput'
 import Label from '@santiment-network/ui/Label'
-import { ASSETS_FILTERS, ETH_WALLET_METRIC } from '../../../utils/constants'
-import { TriggerProjectsSelector } from './TriggerProjectsSelector'
 import {
-  ALL_ERC20_PROJECTS_QUERY,
-  allProjectsForSearchGQL
-} from '../../../../../pages/Projects/allProjectsGQL'
+  METRIC_TARGET_OPTIONS,
+  METRIC_TARGET_ASSETS,
+  METRIC_TARGET_WATCHLIST
+} from '../../../utils/constants'
+import GetWatchlists from '../../../../Watchlists/GetWatchlists'
+import GetProjects from '../../../common/projects/getProjects'
+import { mapToAssets } from '../../../utils/utils'
 import styles from '../signal/TriggerForm.module.scss'
-import { ASSETS_BY_WALLET_QUERY } from '../../../../HistoricalBalance/common/queries'
-import {
-  mapAssetsHeldByAddressToProps,
-  mapToAssets,
-  isPossibleEthAddress
-} from '../../../utils/utils'
-
-const isDisabledWalletAddressField = (
-  canUseMappedErc20,
-  target,
-  allErc20Projects
-) => {
-  if (canUseMappedErc20) {
-    return false
-  }
-
-  return (
-    !target ||
-    !target.value ||
-    !(
-      target.value === 'ethereum' ||
-      allErc20Projects.find(x => x.value === target.value)
-    )
-  )
-}
 
 const propTypes = {
-  metaFormSettings: PropTypes.any,
-  metric: PropTypes.any.isRequired,
-  target: PropTypes.any,
-  setFieldValue: PropTypes.func.isRequired,
-  byAddress: PropTypes.string,
-  assets: PropTypes.array
+  metaFormSettings: PropTypes.any
 }
 
 const TriggerFormAssetWallet = ({
-  data: { allErc20Projects = [], allProjects = [] } = {},
-  target,
   metaFormSettings,
-  metric,
-  assets = [],
-  setFieldValue,
-  byAddress = ''
+  values: { signalType }
 }) => {
   const defaultSignalType = metaFormSettings.signalType
-  const isEthWallet = metric.value === ETH_WALLET_METRIC.value
 
-  const [erc20List, setErc20] = useState(allErc20Projects)
-  const [allList, setAll] = useState(allProjects)
-  const [heldAssets, setHeldAssets] = useState(assets)
+  const { target: defaultAsset } = metaFormSettings
 
-  useEffect(() => {
-    allErc20Projects.length && setErc20(allErc20Projects)
-    allProjects.length && setAll(allProjects)
-    assets.length && setHeldAssets(assets)
-  })
-
-  const canUseMappedErc20 = !!byAddress && assets.length > 0
-  const disabledWalletField = isDisabledWalletAddressField(
-    canUseMappedErc20,
-    target,
-    erc20List
-  )
-  const selectableProjects =
-    canUseMappedErc20 && assets.length > 0 ? assets : allList
+  const isAsset = signalType.value === METRIC_TARGET_ASSETS.value
+  const isWatchlist = signalType.value === METRIC_TARGET_WATCHLIST.value
```

Comment:
`[DM-1-2]` `signalType.value` is enum. If `isAsset` is true, then `isWatchlist` check is redundant.